src/shared/FocusTrap/hooks/useFocusTrap.ts

```diff
@@ -5,18 +5,23 @@ import { eventKeys } from '../../utilities/eventKeys';
 const SELECTORS =
     'a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"], iframe, object, embed';
 
+const INITIAL_FOCUS_DELAY = 100;
+
 export function useFocusTrap(
     visible = true
 ): React.MutableRefObject<HTMLDivElement> {
     const elRef = useRef<any>(null);
-    const handleFocus = (e: React.KeyboardEvent) => {
-        const focusableEls = [
-            ...elRef.current.querySelectorAll(SELECTORS),
-        ].filter(
+    const restoreFocusRef = useRef<any>(null);
+
+    const getFocusableElements = () => {
+        return [...elRef.current.querySelectorAll(SELECTORS)].filter(
             (el) =>
                 !el.hasAttribute('disabled') && !el.getAttribute('aria-hidden')
         );
+    };
 
+    const handleFocus = (e: React.KeyboardEvent) => {
+        const focusableEls = getFocusableElements();
         const firstFocusableEl = focusableEls[0];
         const lastFocusableEl = focusableEls[focusableEls.length - 1];
```

Comment:
If there are no focuable elemets in the modal wont this break, lets handle this as well